src/platform/backends/lxd/lxd_request.cpp

```diff
@@ -135,3 +150,48 @@ const QJsonObject mp::lxd_request(mp::NetworkAccessManager* manager, const std::
 
     return lxd_request_common(method, url, timeout, handle_request);
 }
+catch (const LXDRuntimeError& e)
+{
+    mpl::log(mpl::Level::error, request_category, e.what());
+
+    throw;
+}
+
+const QJsonObject mp::lxd_wait(mp::NetworkAccessManager* manager, const QUrl& base_url, const QJsonObject& task_data,
+                               int timeout) // clang-format off
+try // clang-format on
+{
+    QJsonObject task_reply;
+
+    if (task_data["metadata"].toObject()["class"] == QStringLiteral("task") &&
+        task_data["status_code"].toInt(-1) == 100)
+    {
+        QUrl task_url(QString("%1/operations/%2/wait")
+                          .arg(base_url.toString())
+                          .arg(task_data["metadata"].toObject()["id"].toString()));
+
+        task_reply = lxd_request(manager, "GET", task_url, mp::nullopt, timeout);
+
+        if (task_reply["error_code"].toInt() >= 400)
+        {
+            throw mp::LXDRuntimeError(fmt::format("Error waiting on operation: {}", task_reply["error"].toString()));
+        }
+        else if (task_reply["status_code"].toInt() == 400)
+        {
+            throw mp::LXDRuntimeError(fmt::format("Failure waiting on operation: {}", task_reply["status"].toString()));
+        }
+        else if (task_reply["metadata"].toObject()["status_code"].toInt() == 400)
+        {
+            throw mp::LXDRuntimeError(
+                fmt::format("Operation completed with error: {}", task_reply["metadata"].toObject()["err"].toString()));
+        }
```

Comment:
Should all these not be `!= 200`? Like, It's an error any time the returned status is not `200`? I fail to see documentation on specific status codes returned by `/wait`…

Reply:
Ugh...so LXD has like three different ways to report errors and status. 

So if there is an HTTP error, then `task_reply["error_code"]` is going to be set and per the documentation will "be one of of 400, 401, 403, 404, 409, 412 or 500."

As for the API call itself, there is also a status code which is obtained via `task_reply["status_code"]`.  According to the docs, a failure is **only** indicated by code 400.

Then there is the associated operation of the API call which is obtained via `task_reply["metadata"].toObject()["status_code"]` and again, a failure is only indicated by code 400.

I got these values via https://lxd.readthedocs.io/en/latest/rest-api/#return-values and https://lxd.readthedocs.io/en/latest/rest-api/#status-codes.

Also, no we don't want it to be `!= 200`.  For example, if an operation is running, it will have a status code of `103` and we don't want to fail on that.  To expand that even further, we don't want to fail on any status code in the 100's or 200.

Reply:
I just worry LXD might introduce a new error/status code and we'll miss it. Should everywhere be `>= 400`, then, as LXD tightly follows HTTP status codes?

Reply:
> I just worry LXD might introduce a new error/status code and we'll miss it.

Well, I think we're going to have to trust the docs here.  If they deviate from what that says, then they have broken the API and if they do things properly, we should be made aware of that.

Reply:
As discussed, I now have `>= 400` for the two conditions and now also log the error/status code.